Bug fix on records with 4 or more modifiers

Bjarno opened the pull request:
This pull request fixes an issue with record definitions that have 4 or more modifiers defined within them.

```
#lang r7rs
(import (scheme base))
(define-record-type test
  (make a b c d)
  test?
  (a get-a a!)
  (b get-b b!)
  (c get-c c!)
  (d get-d d!))
```

This program currently raises the following error:

```
make-struct-field-mutator: arity mismatch;
 the expected number of arguments does not match the given number
  given: 6
```

Upon investigation in `r7rs-lib/private/records.rkt`, I identified that the call to `make-struct-field-mutator` (which generates the mutator procedure) contains an expansion of `'field-modifiers` in such a way that when there are more modifiers, `make-struct-field-mutator` is called with too many arguments. From what I can tell, from reading the documentation on `make-struct-field-mutator`, only the name of the modifier that is to be generated has to be supplied. This commit does exactly that.

I haven't yet fully tested whether this breaks other parts, but it seems to be correct to get the small example program listed above to work.

Tagging @svdvonde and @endymion64 (who provided the example code above), who raised this issue to me.

Commit:
Fixes reading of characters in the repl

Bjarno committed:
Merge branch 'lexi-lambda:master' into master

Commit:
Bug fix for generation of mutators in `define-record-type`.
The `'field-modifiers` was erronously expanded in the call to `make-struct-field-mutator`.

Bjarno commented:
I might have pulled/synced my fork incorrectly, so github also lists a commit from an earlier pull request (#19) I made in this one. My apologies.

svdvonde commented:
I'm running into this issue with records in a class we're teaching at uni using this R7RS package. 

@lexi-lambda Do you have some time to look at the proposed fix?

svdvonde commented:
@lexi-lambda Second semester at our uni is about to start. Can I try again to get this merged?

Commit:
Minor bug fix to improve debugging experience.
The wrong names where shown for struct accessors/mutators when displaying in (e.g.,) the REPL.

mflatt commented:
@lexi-lambda ping - thanks!
